justifi-tech/web-component-library
# Add resize method to CardForm and BankAccountForm

Add `resize` method to `CardForm` and `BankAccountForm` which will send a post message `resize` to the iframed `payments-js` application


Links
-----
Related payments-js PR: https://github.com/justifi-tech/payments-js/pull/97

Developer considerations
--------------

  - If this is a MAJOR version, what are the breaking API changes? Was anything added to the API?
  - If this is a MINOR version, what feature was added? Why would consumers want this new version?
  - If this is a PATCH version, what was fixed?


Steps for Testing/QA
--------------------
- Pull the corresponding `payments-js` branch and `yarn run` (https://github.com/justifi-tech/payments-js/pull/97)
- Pull this branch
- Temporarily change `productionIframeOrigin` in `payment-method-form.tsx` to point to local payments-js (http://localhost:3003/v2)
- Run `yarn start` from the `stencil-library` directory
-  Go to http://localhost:6006/?path=/story/components-cardform--basic
- Click 'Validate' to cause validation messages to appear. Shrink the window so that the validation messages wrap, which should cause things to be cut off because payments-js does not know to tell the component to resize it's iframe
- Click 'Resize' and the issue of things being cut off should be fixed
- Sign off on the corresponding PR which includes a passing regression test suite: https://github.com/justifi-tech/payments-js/pull/97
